src/plugins/expressions/common/execution/execution.ts

```diff
@@ -139,8 +142,8 @@ export class Execution<
 
   public readonly expression: string;
 
-  public get result(): Promise<Output | ExpressionValueError> {
-    return this.firstResultFuture.promise;
+  public get result(): Observable<Output | ExpressionValueError> {
```

Comment:
I know it was this way before, but why is `ExpressionValueError` returned and not thrown?

Reply:
That was like that before, but the return type was incorrect. In case of error during the function invocation, we are returning an object containing `{ type: 'error' }`. That's how it was done before. I had to reimplement the existing behavior to keep it backward compatible.

Reply:
the way expressions were designed they should always return an output (possibly of type "error") which is then passed to appropriate renderer which might then throw. the reason to do it so was to allow execution of expressions (partially) on the server and we wanted to serialize errors and send them over to the client.